sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -812,9 +876,11 @@ export class Sequence implements SeqJson {
         ? {
             // @ts-ignore : 'Step' found in JSON Spec
             steps: json.steps.map((c: Step) => {
-              if (c.type === 'command') return CommandStem.fromSeqJson(c as CommandStem, localNames, parameterNames);
-              else if (c.type === 'ground_block') return Ground_Block.fromSeqJson(c as Ground_Block);
-              else if (c.type === 'ground_event') return Ground_Event.fromSeqJson(c as Ground_Event);
+              if (c.type === StepType.Command) return CommandStem.fromSeqJson(c as CommandStem, localNames, parameterNames);
+              else if (c.type === StepType.GroundBlock) return Ground_Block.fromSeqJson(c as Ground_Block);
+              else if (c.type === StepType.GroundEvent) return Ground_Event.fromSeqJson(c as Ground_Event);
+              else if (c.type === StepType.Activate) return ActivateStep.fromSeqJson(c as ActivateStep);
+              else if (c.type === StepType.Load) return LoadStep.fromSeqJson(c as LoadStep);
```

Comment:
Switching this to a switch might read a little cleaner, we can also do this [trick](https://www.meticulous.ai/blog/safer-exhaustive-switch-statements-in-typescript) to make sure capture all the types.

Reply:
I like the exhaustedGuard approach, but we can't use it here because the default: value is a SeqJson that doesn't have an eDSL. Instead, we'll pass it along. This is in case there is a new step that doesn't have an eDSL class yet. This will allow the user to embed SeqJson until we come up with an eDSL for that step.

I will see if I can use it elsewhere in the preface though.

Reply:
Ah, yeah that makes sense!